### PR TITLE
feat(refactor configs): refactor config into separate crate, and fix infinite loop bug in export

### PR DESCRIPTION
**Pull request by JoeCap08055**
Refactor configuration data (chain constants, schema ids & schemas) into their own independent crate

# Checklist
- [x] Tests added

### CODE REVIEW

File: config/resources/configs/frequency.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+  "maxGraphPageSizeBytes": 2048,
```
Comment (aramikm) [inline config/resources/configs/frequency.json:2]: I think this should be 1024

File: config/resources/configs/frequency.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  "maxPageId": 16,
+  "maxKeyPageSizeBytes": 1024,
```
Comment (aramikm) [inline config/resources/configs/frequency.json:4]: This is higher around 64k

File: config/resources/configs/frequency-rococo.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  "maxGraphPageSizeBytes": 2048,
```
Comment (aramikm) [inline config/resources/configs/frequency-rococo.json:2]: 1024

File: config/resources/configs/frequency-rococo.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  "maxPageId": 16,
+  "maxKeyPageSizeBytes": 1024,
```
Comment (aramikm) [inline config/resources/configs/frequency-rococo.json:4]: 64k if I'm not wrong

File: config/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+			.unwrap();
+	pub static ref PUBLIC_GRAPH_SCHEMA: Schema =
+		Schema::parse_str(include_str!("../resources/schemas/public_graph.json")).unwrap();
+	pub static ref PRIVATE_GRAPH_CHUNK_SCHEMA: Schema =
+		Schema::parse_str(include_str!("../resources/schemas/user_private_graph_chunk.json"))
+			.unwrap();
+}
+
+/// Privacy Type of the graph
+#[derive(Clone, Copy, PartialEq, Ord, Eq, PartialOrd, Debug, Hash, Serialize, Deserialize)]
```
Comment (aramikm) [inline config/src/lib.rs:26]: Should these types be defined out side config? any intentions why here?

File: core/src/graph/graph.rs [hunk 6]
```diff
@@ -603,4 +609,17 @@ mod test {
 		graph_connections.sort();
 		assert_eq!(test_connections, graph_connections);
 	}
+
+	#[test]
+	#[timeout(5000)] // found an infinite loop bug, so let's make sure this terminates successfully
```
Comment (aramikm) [inline core/src/graph/graph.rs:614]: interesting, should check our loop conditions

File: config/resources/configs/frequency-rococo.json (hunk 3 above)
Comment (JoeCap08055) [reply config/resources/configs/frequency-rococo.json:2]: right; I put in placeholders & then forgot to check in the frequency repo 🤦🏻

File: config/src/lib.rs (hunk 5 above)
Comment (JoeCap08055) [reply config/src/lib.rs:26]: In order to be able to have config manage the mapping ConnectionType <=> SchemaId, it needs to know about those enums. If we define them in the core crate, then we get a circular dependency. The other option would be to factor out into a `dsnp-graph-common` or `dsnp-graph-types` crate...

Comment (aramikm) [reply config/src/lib.rs:26]: I see, let's keep it here then.

File: config/resources/configs/frequency-rococo.json (hunk 4 above)
Comment (JoeCap08055) [reply config/resources/configs/frequency-rococo.json]: Updated to match frequency repo

File: config/resources/configs/frequency.json (hunk 1 above)
Comment (JoeCap08055) [reply config/resources/configs/frequency.json]: Updated to match frequency repo

File: config/resources/configs/frequency.json (hunk 2 above)
Comment (JoeCap08055) [reply config/resources/configs/frequency.json]: Updated to match frequency repo

**Review by aramikm — APPROVED** [review]
Looks good!

**Pull request merged**